addons/stock/models/product.py

```diff
@@ -114,7 +114,10 @@ def _product_available(self, field_names=None, arg=False):
     @api.multi
     def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False, to_date=False):
         domain_quant_loc, domain_move_in_loc, domain_move_out_loc = self._get_domain_locations()
-        domain_quant = [('product_id', 'in', self.ids)] + domain_quant_loc
+        if 'location' in self._context:
+            domain_quant = [('location_id.id', '=', self._context['location']), ('product_id', 'in', self.ids)]
+        else:
+            domain_quant = [('product_id', 'in', self.ids)] + domain_quant_loc
```

Comment:
Hi you can just do that:

domain_quant = [('product_id', 'in', self.ids)] + (self.env.context.get('location_id', False) or domain_quant_loc)